### PR TITLE
Code of conduct refresh

### PR DESCRIPTION
**Pull request by robcrowley**
PR containing refreshed code of conduct based on the great work done the JS Conf AU team.

### CODE REVIEW

File: pages/code-of-conduct.tsx [hunk 1]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
 
+    <h2>Purpose</h2>
     <p>
-      All attendees, speakers, sponsors and volunteers at our conference are required to agree with the following code
-      of conduct. We are expecting cooperation from all participants to help ensuring a safe, welcoming and inclusive
-      environment for everybody.
+      {props.pageMetadata.conference.Name} is dedicated to providing a harassment-free conference experience for
+      everyone, regardless of gender, gender identity and expression, sexual orientation, disability, physical
+      appearance, body size, race, age or religion. We do not tolerate harassment of conference participants in any
+      form. Sexual language and imagery is not appropriate for any conference venue, including talks.
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: The "for any conference venue" bit reads weirdly?

File: pages/code-of-conduct.tsx [hunk 2]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
-    <h2>The Less Quick Version</h2>
+    <h2>Reporting</h2>
+
     <p>
-      Harassment includes offensive verbal comments related to gender, gender identity and expression, age, sexual
-      orientation, disability, physical appearance, body size, race, ethnicity, religion, sexual images in public
-      spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of
-      talks or other events, inappropriate physical contact, and unwelcome sexual attention.
+      If someone makes you or anyone else feel unsafe or unwelcome, please report it as soon as possible. Conference
+      staff can be identified by event branded t-shirts. Harassment and other Code of Conduct violations reduce the
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: green, event-branded tshirts and organiser name tags?

File: pages/code-of-conduct.tsx [hunk 3]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
+      follow up an anonymous report with you directly, but we will fully investigate it and take whatever action is
+      necessary to prevent a recurrence.
+    </p>
+
+    <h3>Personal report</h3>
+
+    <p>You can make a personal report by:</p>
+
+    <ul>
+      <li>Contacting a staff member, identified by event branded t-shirts</li>
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: Whatever we put above should also be put here, might make sense as a variable in the Conference class

File: pages/code-of-conduct.tsx [hunk 4]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
+      <li>Sending a direct message to our Twitter account: {props.pageMetadata.conference.Socials.Twitter}</li>
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: You can create a link that initiates a DM to that account

File: pages/code-of-conduct.tsx [hunk 5]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
+      <li>Emailing us: {props.pageMetadata.conference.Socials.Email}</li>
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: Make this a mailto: link, maybe with a subject pre-populated?

File: pages/code-of-conduct.tsx [hunk 6]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
+      <li>Non Emergency: {props.pageMetadata.conference.ImportantContactNumbers.NonEmergencyMedical}</li>
+    </ul>
+
+    <h2>Photo Policy</h2>
+
+    <p>We will have colored lanyards for attendees to indicate their comfort level with being photographed:</p>
+
+    <ul>
+      <li>Blue: ask beforehand</li>
+      <li>Black: fine to photograph</li>
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: Make this green, since we'll use our green lanyards

File: pages/code-of-conduct.tsx [hunk 7]
```diff
@@ -10,89 +10,160 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
-        </em>
-      </small>
+      For more information please familiarise yourself with{' '}
+      <a href="http://geekfeminism.wikia.com/wiki/Nonsexist_language">Geek Feminism wiki guide</a>.
     </p>
+
+    <h2>Attribution</h2>
+
+    <p>
+      This Code of Conduct was based on{' '}
```
Comment (robdmoore) [inline pages/code-of-conduct.tsx]: Should we reference DDD Melbourne's too?

File: pages/code-of-conduct.tsx (hunk 1 above)
Comment (ucichr) [reply pages/code-of-conduct.tsx]: "Sexual language and imagery is not appropriate anywhere. This applies to talks also.

Comment (ucichr) [reply pages/code-of-conduct.tsx]: "regardless of but not limited to" ? Not sure it's really required though.

File: pages/code-of-conduct.tsx [hunk 8]
```diff
@@ -10,89 +10,175 @@ export default withPageMetadata((props: WithPageMetadataProps) => (
...
+
     <p>
-      Email <a href={props.pageMetadata.conference.ContactEmail}>{props.pageMetadata.conference.ContactEmail}</a> or
-      call our designated contact person {props.pageMetadata.conference.EmergencyContactName} on{' '}
-      {props.pageMetadata.conference.EmergencyContactPhoneNumber}. You can also approach any one of the conference staff
-      who will be identifiable by the branded DDD Perth t-shirts they'll be wearing.
+      If a participant engages in harassing behaviour, event organisers retain the right to take any actions to keep the
+      event a welcoming environment for all participants. This includes warning the offender or expulsion from the
+      conference with no refund.
     </p>
```
Comment (ucichr) [inline pages/code-of-conduct.tsx]: DMs are open on twitter. You may wish to note who sees them, being Rob and I. It's useful I think to provide an option that is seen by a woman.
Also noting that any contact regarding the CoC will be treated discreetly.

File: pages/code-of-conduct.tsx (hunk 4 above)
Comment (ucichr) [reply pages/code-of-conduct.tsx]: I would also point out who sees these DMs. Particularly as it's one of the ways you can report to a woman (me).

File: pages/code-of-conduct.tsx (hunk 5 above)
Comment (ucichr) [reply pages/code-of-conduct.tsx]: Same comment on who sees the email.

File: pages/code-of-conduct.tsx (hunk 1 above)
Comment (robcrowley) [reply pages/code-of-conduct.tsx]: Yes, agree. I've gone with Rebecca's suggestion "Sexual language and imagery is not appropriate anywhere. This applies to talks also."

File: pages/code-of-conduct.tsx (hunk 2 above)
Comment (robcrowley) [reply pages/code-of-conduct.tsx]: Works for me.

File: config/conference.ts [hunk 9]
```diff
@@ -138,6 +141,13 @@ const Conference: IConference = {
     { Url: '/static/images/strip/5.jpg', Alternate: 'Donna Edwards speaking at DDD 2017' },
   ],
 
+  ImportantContactNumbers: {
+    Police: '2 Fitzgerald St, Northbridge WA 6003 ph: (08) 9422 7111',
+    CentreAgainstSexualAssault: '1800 806 292',
+    EmergencyMedical: '197 Wellington St, Perth WA 6000',
```
[the remark this answers is not in the document]
Comment (robcrowley) [reply config/conference.ts]: Thanks for spotting this. I did intend to include the name of the hospital / medical centre.

File: pages/code-of-conduct.tsx (hunk 2 above)
Comment (robcrowley) [reply pages/code-of-conduct.tsx]: I went with "event-branded tshirts and organiser name tags". Adding the t-shirt color made it specific to DDD Perth and as such would need to be parameterised to be of value to other conferences. I am fine with that but not sure of the value of adding the colour is. Conference staff are pretty prominent all event long.

**Pull request merged**